RFCs/0009-Fn::ForEach.md

```diff
@@ -0,0 +1,1461 @@
+    Properties:
+      SubnetId: !Ref TransitSubnetC
+      NetworkAclId: !Ref NaclTransitSubnet
+  NaclPublicSubnet:
+    Type: 'AWS::EC2::NetworkAcl'
+    Properties:
+      VpcId: !Ref VPC
+  PublicSubnetA:
+    Type: 'AWS::EC2::Subnet'
+    Properties:
+      VpcId: !Ref VPC
+  NaclPublicSubnetAAssociation:
+    Type: 'AWS::EC2::SubnetNetworkAclAssociation'
+    Properties:
+      SubnetId: !Ref PublicSubnetA
+      NetworkAclId: !Ref NaclPublicSubnet
+  PublicSubnetB:
+    Type: 'AWS::EC2::Subnet'
+    Properties:
+      VpcId: !Ref VPC
+  NaclPublicSubnetBAssociation:
+    Type: 'AWS::EC2::SubnetNetworkAclAssociation'
+    Properties:
+      SubnetId: !Ref PublicSubnetB
+      NetworkAclId: !Ref NaclPublicSubnet
+  PublicSubnetC:
+    Type: 'AWS::EC2::Subnet'
+    Properties:
+      VpcId: !Ref VPC
+  NaclPublicSubnetCAssociation:
+    Type: 'AWS::EC2::SubnetNetworkAclAssociation'
+    Properties:
+      SubnetId: !Ref PublicSubnetC
+      NetworkAclId: !Ref NaclPublicSubnet
+```
+
+### Fn::ForEach in Properties section
+
+#### Use case 4: Replicate similar properties
+
+In this example, customer is using Fn::ForEach to duplicate same inputs to  `AWS::EC2::Instance` properties such as ImageId, InstanceType, and AvailabilityZone.
+
+#### JSON
+
+```json
+{
+    "AWSTemplateFormatVersion": "2010-09-09",
+    "Transform": "AWS::LanguageExtensions",
+    "Mappings": {
+        "InstanceA": {
+            "Properties": {
+                "ImageId": "ami-id1",
+                "InstanceType": "m5.xlarge"
+            }
+        },
+        "InstanceB": {
+            "Properties": {
+                "ImageId": "ami-id2"
+            }
+        },
+        "InstanceC": {
+            "Properties": {
+                "ImageId": "ami-id3",
+                "InstanceType": "m5.2xlarge",
+                "AvailabilityZone": "us-east-1a"
+            }
+        }
+    },
+    "Resources": {
+        "Fn::ForEach::Instances": [
+            "InstanceLogicalId",
+            ["InstanceA", "InstanceB", "InstanceC"],
+            {
+                "${InstanceLogicalId}": {
+                    "Type": "AWS::EC2::Instance",
+                    "Properties": {
+                        "DisableApiTermination": true,
+                        "UserData": {
+                            "Fn::Base64": {
+                                "Fn::Join": ["", [
+                                    "#!/bin/bash\n",
+                                    "yum update -y\n",
+                                    "yum install -y httpd.x86_64\n",
+                                    "systemctl start httpd.service\n",
+                                    "systemctl enable httpd.service\n",
+                                    "echo \"Hello World from $(hostname -f)\" > /var/www/html/index.html\n"
+                                ]]
+                            }
+                        },
+                        "Fn::ForEach::Properties": [
```

Comment:
Can `Fn::ForEach` in the Properties section also be used when the resource itself wasn't generated by an  `Fn::ForEach`? Can't think of a good use case off-hand, but wanted to hear if we're planning to restrict it.